contracts/amm/AmmDataProvider.sol

```diff
@@ -1,25 +1,31 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
 pragma solidity 0.8.0;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
 import "./IAmmDataProvider.sol";
 import "../token/ISimpleToken.sol";
 import "../series/ISeriesController.sol";
+import "../series/IPriceOracle.sol";
 import "../series/SeriesLibrary.sol";
 import "../libraries/Math.sol";
 
 contract AmmDataProvider is IAmmDataProvider {
     ISeriesController public seriesController;
     IERC1155 public erc1155Controller;
+    IPriceOracle public priceOracle;
 
     event AmmDataProviderCreated(
         ISeriesController seriesController,
-        IERC1155 erc1155Controller
+        IERC1155 erc1155Controller,
+        IPriceOracle priceOracle
     );
 
     constructor(
         ISeriesController _seriesController,
-        IERC1155 _erc1155Controller
+        IERC1155 _erc1155Controller,
+        IPriceOracle _priceOracle
```

Comment:
Still reviewing logic of code but do we need to add the price oracle to the deploy script in
``scripts/lib/deploy_singleton_contracts.ts``? Because I don't think we currently have it in there.

Reply:
This is resolved with the added line in the scripts folder.